### PR TITLE
increase timeout to 30s

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
We're seeing frequent context timeout errors during `GetObject` API calls. This PR increases the default timeout from 5 to 30 seconds.

Error:
```
RequestCanceled: request context canceled
caused by: context deadline exceeded
```
### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

**Review by mcastorina — APPROVED** [review]

**Comment by rosecodym** [comment]
Are we sure that the previous timeout is wrong? My concern is that these timeouts were caused by something other than a bunch of large objects, so if we do this then we won't scan any more data but each scan will take six times as long.

**Comment by ahrav** [comment]
> Are we sure that the previous timeout is wrong? My concern is that these timeouts were caused by something other than a bunch of large objects, so if we do this then we won't scan any more data but each scan will take six times as long.

According to the documentation for the library we're using:
```
// GetObject API operation for Amazon Simple Storage Service.
//
// Retrieves an object from Amazon S3.
```

I _believe_ the timeout was initially set under the assumption that we were only retrieving metadata, but the docs confirm it retrieves the entire object. I'm not certain what else might be causing the context deadline.

**Comment by ahrav** [comment]
Adjusted the logger to use our ctx. I also included the object size in the logger kvp.

File: pkg/sources/s3/s3.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ func (s *Source) setMaxObjectSize(maxObjectSize int64) {
 func (s *Source) newClient(region, roleArn string) (*s3.S3, error) {
 	cfg := aws.NewConfig()
 	cfg.CredentialsChainVerboseErrors = aws.Bool(true)
+	cfg.LogLevel = aws.LogLevel(aws.LogDebugWithRequestErrors)
```
Comment (mcastorina) [inline pkg/sources/s3/s3.go:134]: Was this just for testing or are we keeping it in?

File: pkg/sources/s3/s3.go [hunk 2]
```diff
@@ -271,46 +272,63 @@ func (s *Source) getRegionalClientForBucket(ctx context.Context, defaultRegionCl
...
+	page *s3.ListObjectsV2Output,
+	errorCount *sync.Map,
+	pageNumber int,
+	objectCount *uint64,
+) {
+	for _, o := range page.Contents {
+		if o == nil {
+			continue
 		}
+		obj := *o
```
Comment (mcastorina) [inline pkg/sources/s3/s3.go:289]: Why do we need to make a shallow copy here?

**Review by mcastorina — APPROVED** [review]
This seems good, just had a few non-blocking questions.

One other thought: can `log` be removed from the struct now, or is it still referenced in other parts? Not saying it should be removed in this PR, but I saw you corrected many of the log calls to use the passed `ctx`.

**Comment by ahrav** [comment]
> This seems good, just had a few non-blocking questions.
> 
> One other thought: can `log` be removed from the struct now, or is it still referenced in other parts? Not saying it should be removed in this PR, but I saw you corrected many of the log calls to use the passed `ctx`.

Unfortunately, it's used across the rest of the source 😢 I can clean it up in a separate PR.

File: pkg/sources/s3/s3.go (hunk 2 above)
Comment (ahrav) [reply pkg/sources/s3/s3.go:289]: Oh whoops, I forgot to remove this. I was testing something. 😅 Thanks.

File: pkg/sources/s3/s3.go (hunk 1 above)
Comment (ahrav) [reply pkg/sources/s3/s3.go:134]: I thought it would be useful even for non-debugging purposes since it would only log during errors. What do you think?

Comment (mcastorina) [reply pkg/sources/s3/s3.go:134]: Yeah, makes sense to me.

**Pull request merged**